### PR TITLE
[BAAS-90] add index API

### PR DESCRIPTION
**Pull request by ivy-dong**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -145,7 +145,35 @@ Also you need to confirm the `sign-util` has the executable permission, when you
 
 ```
 
+### Index set 
+```java
+        String privatekeyBase64 = "mKyNuvcWrE5ZtverSYVjxu4LSTDlnLkmF/qvYeq0hU6kEsJKGAZb1CkEFE9qxMytNGPXyIy8gekAdB1rIaVNzQ==";
```
Comment (ewangplay) [inline README.md]: Index Set API don't need these params, pls remove it.

File: README.md [hunk 2]
```diff
@@ -145,7 +145,35 @@ Also you need to confirm the `sign-util` has the executable permission, when you
...
+
+        String response = wallet.IndexSet(jsonheader, jsondata);
+
+```
 
+### Index get 
+```java
+        String privatekeyBase64 = "mKyNuvcWrE5ZtverSYVjxu4LSTDlnLkmF/qvYeq0hU6kEsJKGAZb1CkEFE9qxMytNGPXyIy8gekAdB1rIaVNzQ==";
+        String nonce = "nonce";
+        String created = "1526613187";
```
Comment (ewangplay) [inline README.md]: same as above

File: README.md [hunk 3]
```diff
@@ -145,7 +145,35 @@ Also you need to confirm the `sign-util` has the executable permission, when you
...
+
+        String strdata = "{\"id\":\"did:axn:64ec602c-c9e5-4835-8e63-c6f0a619805c\",\"indexs\":{\"combined_index\":[\"first-me\",\"second-me\",\"third-meu\"],\"individual_index\":[\"individual-me-001\",\"individual-me-002\",\"individual-me-003\"]}}";
+        JSONObject jsondata = JSON.parseObject(strdata);
+
+        JSONObject jsonheader = JSON.parseObject(strheader);
+
+        String response = wallet.IndexSet(jsonheader, jsondata);
```
Comment (ewangplay) [inline README.md]: should be IndexGet

File: src/main/java/com/arxanfintech/sdk/wallet/Wallet.java [hunk 4]
```diff
@@ -391,4 +391,39 @@ private String ProcessTx(JSONObject jsonheader, JSONObject jsonbody) {
...
         }
     }
+
+    public String IndexSet(JSONObject jsonheader, JSONObject jsonbody) {
+        Request request = new Request();
+        request.client = this.client;
+        request.body = jsonbody;
+        request.header = jsonheader;
+        request.crypto = crypto;
+        request.url = "http://" + request.client.Address + "/wallet-ng/v1/index/set";
```
Comment (ewangplay) [inline src/main/java/com/arxanfintech/sdk/wallet/Wallet.java]: You cannot assume that the wallet service must be accessed through the wasabi gateway, so should not hard code the 'wallet-ng' route tag at here. pls refer the wallet-sdk-go to set route tag.

File: src/main/java/com/arxanfintech/sdk/wallet/Wallet.java [hunk 5]
```diff
@@ -391,4 +391,39 @@ private String ProcessTx(JSONObject jsonheader, JSONObject jsonbody) {
...
+        }
+    }
+
+    public String IndexGet(JSONObject jsonheader, JSONObject jsonbody) {
+        Request request = new Request();
+        request.client = this.client;
+        request.body = jsonbody;
+        request.header = jsonheader;
+        request.crypto = crypto;
+        request.url = "http://" + request.client.Address + "/wallet-ng/v1/index/get";
```
Comment (ewangplay) [inline src/main/java/com/arxanfintech/sdk/wallet/Wallet.java]: same as above

File: README.md (hunk 1 above)
Comment (ivy-dong) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (ivy-dong) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (ivy-dong) [reply README.md]: done

File: src/main/java/com/arxanfintech/sdk/wallet/Wallet.java (hunk 4 above)
Comment (ivy-dong) [reply src/main/java/com/arxanfintech/sdk/wallet/Wallet.java]: done

File: src/main/java/com/arxanfintech/sdk/wallet/Wallet.java (hunk 5 above)
Comment (ivy-dong) [reply src/main/java/com/arxanfintech/sdk/wallet/Wallet.java]: done

File: src/main/java/com/arxanfintech/sdk/wallet/Wallet.java [hunk 6]
```diff
@@ -57,7 +62,7 @@ public String Register(JSONObject jsonheader, JSONObject jsonbody) throws Except
         request.body = jsonbody;
         request.header = jsonheader;
         request.crypto = crypto;
-        request.url = "http://" + request.client.Address + "/wallet-ng/v1/wallet/register";
+        request.url = "http://" + request.client.Address + "/" + request.client.RouteTag + "/v1/wallet/register";
```
Comment (ewangplay) [inline src/main/java/com/arxanfintech/sdk/wallet/Wallet.java]: 这样设置RouteTag还是不对的，我的意思是RouteTag就不应该包含在请求的url中。原因是：一旦把RouteTag包含在了url中，就相当于假定了一定要通过wasabi网关来访问wallet-ng服务。那么我不经过wasabi网关而直接访问wallet-ng服务怎么办？所以正确的方案是：把RouteTag从请求url里去掉，添加到header中(header的Key就是RouteTag），wasabi网关对这个header进行了处理，会读取到这个route tag，又不影响直接访问wallet-ng服务。

**Review by ewangplay — CHANGES_REQUESTED** [review]

File: src/main/java/com/arxanfintech/sdk/wallet/Wallet.java (hunk 6 above)
Comment (ivy-dong) [reply src/main/java/com/arxanfintech/sdk/wallet/Wallet.java]: done

File: README.md [hunk 7]
```diff
@@ -21,7 +21,7 @@ When you use wallet-sdk-java, you should reference project like this:
 <dependency>
     <groupId>com.arxanfintech</groupId>
     <artifactId>wallet-sdk-java</artifactId>
-    <version>2.0.1b</version>
+    <version>2.1test</version>
```
Comment (ewangplay) [inline README.md]: 2.1test? why to name this version?

File: README.md [hunk 8]
```diff
@@ -125,14 +129,14 @@ Also you need to confirm the `sign-util` has the executable permission, when you
 
         JSONObject jsonheader = JSON.parseObject(strheader);
         
-        String response = wallet.CreatePOE(jsonheader, jsondata, did, created, nonce, privatekeyBase64);
```
Comment (ewangplay) [inline README.md:128]: About these ed25519 signature params, is enterprise's or personal's ? It should be clearly commented.

File: README.md (hunk 7 above)
Comment (ivy-dong) [reply README.md]: maven repository (http://mvnrepository.com/) only let me put new version, don't allow update jar. If NOW use 2.1, then the version after test will use another version name. So now use 2.1test just for test.

File: README.md (hunk 8 above)
Comment (ivy-dong) [reply README.md:128]: next pull request will fix this. Complete java doc.

**Review by ewangplay — APPROVED** [review]

**Pull request merged**